Subject: unit1.pas causes an access violation when trying to use an empty
  presetname. This checks for an empty name and avoids accessing it.
Author: Niall Creech <niallcreech@googlemail.com>
Author: Paul Gevers <paul@climbing.nl>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/winff/+bug/521818
Forwarded: not needed

diff -ruN winff-1.2.0/unit1.pas winff-1.2.0.new/unit1.pas
--- winff-1.2.0/unit1.pas	2010-01-26 19:32:08.000000000 +0000
+++ winff-1.2.0.new/unit1.pas	2010-02-21 14:15:08.000000000 +0000
@@ -568,9 +568,11 @@
   sformtop:=getconfigvalue('window/top');
   sformleft:=getconfigvalue('window/left');
 
+  formtop := 0;
   if sformtop <> '' then formtop:=strtoint(sformtop);
   if formtop > 0 then form1.Top := formtop;
 
+  formleft := 0;
   if sformleft <> '' then formleft:=strtoint(sformleft);
   if formleft >0 then form1.Left := formleft;
 
@@ -720,6 +722,9 @@
 catnode : tdomnode;
 category:string;
 begin
+ result := '';
+ if presetname <> '' then
+ begin
    try
     if presets.FindNode(presetname).FindNode('category').HasChildNodes then
     begin
@@ -730,6 +735,7 @@
     category:='';
    end;
    result:=category;
+ end;
 end;
 
 // get the extension of the preset
@@ -760,6 +766,9 @@
 dirnode : tdomnode;
 destdir:string;
 begin
+   result := '';
+ if presetname <> '' then
+ begin
    try
     if presets.FindNode(presetname).FindNode('destdir').HasChildNodes then
       begin
@@ -770,6 +779,7 @@
     destdir:='';
    end;
    result:=destdir;
+ end;
 end;
 
 // save the destination directory to the preset
@@ -977,6 +987,7 @@
 currentpreset:string;
 begin
 currentpreset := getcurrentpresetname(presetbox.Text);
+   destdir := '' ;
 //  destdir:= getpresetdestdir(currentpreset);     // get dest folder from preset
 if destdir <> '' then destfolder.text:= destdir;
 if destfolder.Text='' then destfolder.text := getconfigvalue('general/destfolder');
